src/main.cpp

```diff
@@ -2998,6 +2999,10 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         pos.nTxOffset += ::GetSerializeSize(tx, SER_DISK, CLIENT_VERSION);
     }
 
+    // Derive the various block commitments.
+    auto hashAuthDataRoot = block.BuildAuthDataMerkleTree();
+    auto hashChainHistoryRoot = view.GetHistoryRoot(prevConsensusBranchId);
```

Comment:
Make these `std::optional` defaulting to `nullopt`, and then gate them appropriately (NU5 and Heartwood respectively).